File: Prolog/List/37.drop.pl
% Drop every N'th element from a list.
% Example:
% ?- drop([a,b,c,d,e,f,g,h,i,k],3,X).
% X = [a,b,d,e,g,h,k]


drop([],_,[]).
drop(List,N,Output):- listDropping(List,N,1,Output).

listDropping([],_,_,[]).
listDropping([X|List],N,Counter,Out):- N == Counter, !, listDropping(List,N,1,Out).
listDropping([X|List],N,Counter,[X|Out]):- N1 is Counter+1, listDropping(List,N,N1,Out).

File: Python/List/37.drop.py
def drop(i_list: list,n:int) -> list:
    """
    Drop at multiple of n from the list
    :param n: Drop from the list i_list every N element
    :param i_list: The source list
    :return: The returned list
    """
    assert(n>0)
    _shallow_list = []
    k=1
    for element in i_list:
        if k % n != 0:
            _shallow_list.append(element)
        k+=1
    return _shallow_list

if __name__ == "__main__":
    print(drop([1,2,3,4,5],6))